ga4gh/server/frontend.py

```diff
@@ -729,23 +602,174 @@ def _parseTokenPayload(token):
     return json.loads(decoded_payload)
 
 
-def _mergeCounts(responseObject):
-    table = responseObject['results'].keys()[0]
-    prepare_counts = {}
-    for record in responseObject['results'][table]:
-        for k, v in record.iteritems():
-            if k in prepare_counts:
-                prepare_counts[k].append(Counter(v))
+class FederationResponse(object):
+
+    def __init__(self, request, endpoint, return_mimetype, request_dict):
+        self.results = {}
+        self.status = []
+        self.request = request
+        self.endpoint = endpoint
+        self.return_mimetype = return_mimetype
+        self.request_dict = request_dict
+        self.token, self.access_map = self.handleAccessPermission()
+
+    def handleAccessPermission(self):
+        """
+        user roles are loaded in to the server from the access_list.txt
+        a user-specific access map is generated here based on the id_token and
+        the in-memory server side access map
+
+        if running NoAuth config, user has full access to local datasets
+
+        :return: jwt token, python dict in the form {"project" : "access tier", ...}
+        """
+
+        access_map = {}
+
+        if app.config.get("TYK_ENABLED"):
+            if 'Authorization' in self.request_dict.headers:
+                access_map = {}
+                token = self.request_dict.headers['Authorization']
+                parsed_payload = _parseTokenPayload(token)
+
+                username = parsed_payload.get('preferred_username')
+                if username:
+                    access_map = app.access_map.getUserAccessMap().get(username, {})
+                else:
+                    if app.logger:
```

Comment:
is this equivalent to a one-line `elif`?

Reply:
Ah yep, I think I originally had an 'else' after the app.logger check but I removed it. Could probably rewrite this to elif now

Reply:
nice, makes sense